.github/workflows/scripts/get-codeowners.sh

```diff
@@ -9,15 +9,31 @@
 
 set -euo pipefail
 
+get_component_type() {
+  echo "${COMPONENT}" | cut -f 1 -d '/'
+}
+
 get_codeowners() {
-  echo "$((grep -m 1 "${1}" .github/CODEOWNERS || true) | sed 's/   */ /g' | cut -f3- -d ' ')"
+  echo "$((grep -m 1 "^${1}/" .github/CODEOWNERS || true) | \
```

Comment:
```suggestion
  echo "$((grep -m 1 "^${1}/ " .github/CODEOWNERS || true) | \
```
Sorry, I should have explained my suggestion a little more in-depth. My thought is that if we use the regex `^${1}/ ` to match for the component, we can check that our search (`$1`) is bounded by the start of the line and the CODEOWNERS file delimiting character (a space), which I think should ensure we're not just matching on part of the name.

Do you think if we do this check that we could get away without doing additional validations on the component name below?

Reply:
My apologies, misread the suggestion as simply `/`, not `/ `. Do we have guarantees that indentantions in `CODEOWNERS` will always be a space and not some other whitespace character (like a tab? I'm not seeing anything definitive in GitHub documentation, but our file at least is all spaces after paths from what I can tell.

From testing we can get rid of the extra validation with this change. In my mind it's always nice to validate inputs upon entry, but if we're confident enough with the new grep statement I guess it's fair to remove.

Reply:
That's a fair point, we could just do `^${1}/\s` to capture any other types of whitespace characters. The closest thing I can find to an authoritative guide on the syntax is [here](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax), which uses spaces in their example, but the delimiting character isn't explicitly mentioned.

I'm mostly looking to cut down on code since shell scripting isn't the easiest to read, but if you think we may benefit from additional validation I'll defer to you here.

Reply:
I've added the whitespace regex, thanks!

We can remove the validation for now, then re-add again in the future if we find another case that we need to account for. 👍